lib/datadog/ci/ext/environment.rb

```diff
@@ -324,6 +327,22 @@ def extract_bitrise(env)
           }
         end
 
+        def extract_user_defined_git(env)
+          {
+            Datadog::Ext::Git::TAG_REPOSITORY_URL => env['DD_GIT_REPOSITORY_URL'],
```

Comment:
All these strings should be frozen, migrated to `Datadog::Ext::Git` and prefixed with `ENV_`

e.g. `'DD_GIT_REPOSITORY_URL'` => `Datadog::Ext::Git::ENV_REPOSITORY_URL = 'DD_GIT_REPOSITORY_URL'.freeze`